crates/wasmtime/src/runtime/vm/traphandlers.rs

```diff
@@ -704,11 +868,43 @@ impl CallThreadState {
     /// called. Additionally this isn't safe as it will skip all Rust
     /// destructors on the stack, if there are any.
     #[cfg(has_host_compiler_backend)]
-    unsafe fn unwind(&self) -> ! {
-        debug_assert!(!self.jmp_buf.get().is_null());
-        debug_assert!(self.jmp_buf.get() != CallThreadState::JMP_BUF_INTERPRETER_SENTINEL);
-        unsafe {
-            traphandlers::wasmtime_longjmp(self.jmp_buf.get());
+    unsafe fn unwind(&self, store: &mut dyn VMStore) -> ! {
```

Comment:
Would it be possible to change this to `Option<usize>` representing the payload? That way `UnwindToHost` would assert it's `None` and `UnwindToWasm` would assert it's `Some`.

Reply:
Nah that seems reasonable, this is a feeble attempt by me to future-proof Pulley support a bit more. Let's just defer and deal with it in total later.